extensions/mssql/src/utils.ts

```diff
@@ -20,17 +27,71 @@ export function getAppDataPath() {
 	}
 }
 
-export function getDefaultLogLocation() {
-	if (process.env['VSCODE_DEV'])
+export function removeOldLogFiles(prefix: string) : JSON {
+	return findRemoveSync(getDefaultLogDir(), {prefix: `${prefix}_`,  age: {seconds: getConfigLogRetentionSeconds()}, limit: getConfigLogFilesRemovalLimit()});
+}
+
+export function getConfiguration(config: string = extensionConfigSectionName) : WorkspaceConfiguration {
+	return workspace.getConfiguration(extensionConfigSectionName);
+}
+
+export function getConfigLogFilesRemovalLimit() : number {
+	let config = getConfiguration();
+	if (config) {
+		return Number((config[configLogFilesRemovalLimit]).toFixed(0));
+	}
+	else
 	{
-		return path.join(getAppDataPath(), 'code');
+		return null;
+	}
+}
+
+export function getConfigLogRetentionSeconds() : number {
+	let config = getConfiguration();
+	if (config) {
+		return Number((config[configLogRetentionMinutes] * 60).toFixed(0));
 	}
 	else
 	{
-		return path.join(getAppDataPath(), 'azuredatastudio');
+		return null;
 	}
 }
 
+export function getConfigTracingLevel() : string {
+	let config = getConfiguration();
+	if (config) {
+		return config[configTracingLevel];
+	}
+	else
+	{
+		return null;
```

Comment:
we generally prefer to return `undefined` instead of `null` unless there's a reason we need `null`